common/persistence/sql/plugins/mysql/plugin.go

```diff
@@ -47,12 +51,40 @@ var dsnAttrOverrides = map[string]string{
 	"multiStatements": "true",
 }
 
-// NewSQLDB creates a returns a reference to a logical connection to the
+type plugin struct{}
+
+var _ plugins.Plugin = (*plugin)(nil)
+
+func init() {
+	sql.RegisterPlugin(PluginName, &plugin{})
+}
+
+// InitDB initialize the db object
+func (p *plugin) InitDB(cfg *config.SQL) (plugins.DB, error) {
```

Comment:
consider renaming these InitFoo* methods to NewFoo* methods - its more consistent with the usage here

Reply:
Sounds good